interview-questions.md

```diff
@@ -10,34 +10,121 @@ Then, research the question to expand on your answer. Even if you feel you have
+Action: The edit action retrieves a specific student from the database based on the given ID parameter, and prepares it for rendering in the view. It typically fetches the student record from the model based on the ID, and renders a view for editing the details of that student.
+
 action: "PATCH" location: /students/2
 
+Controller Method: update
+Action: The update action updates a specific student record in the database based on the parameters submitted from a form or API request. It typically validates the input data, finds the student record in the database based on the ID, and updates its attributes before saving it. It may also handle error cases and respond accordingly.
+
 action: "DELETE" location: /students/2
 
+Controller Method: destroy
+Action: The destroy action finds and deletes a specific student record from the database based on the given ID parameter. It typically finds the student record based on the ID, destroys it in the database, and may handle error cases and respond accordingly.
+
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
+    1.  1. As a user, I want to be able to create a new task in my to-do list with a title, description, and due date, so that I can keep track of my tasks and their deadlines.
+
+    2. As a user, I want to be able to view my list of tasks, sorted by their due dates, so that I can easily see which tasks are coming up soon and prioritize my work accordingly.
+
+    3. As a user, I want to be able to mark a task as completed, so that I can keep track of my progress and see which tasks are still pending.
+
+    4. As a user, I want to be able to edit the title, description, or due date of a task, in case I need to update the details or change the deadline.
+
+    5. As a user, I want to be able to delete a task from my to-do list, in case I no longer need to complete it or it was mistakenly added.
+
+    6. As a user, I want to be able to categorize tasks into different categories or tags, so that I can easily filter and organize my tasks based on their types or priorities.
+
+    7. As a user, I want to be able to set reminders or notifications for tasks with upcoming due dates, so that I can be alerted in advance and avoid missing any deadlines.
+
+    8. As a user, I want to be able to search for tasks based on their titles or descriptions, so that I can quickly find specific tasks or keywords within my to-do list.
+
+    9. As a user, I want to be able to prioritize tasks by assigning different levels of importance or urgency, so that I can focus on the most critical tasks first and manage my workload effectively.
+
+    10. As a user, I want to be able to view completed tasks separately from pending tasks, so that I can review my progress and keep track of my completed tasks for reference.
```

Comment:
Amazing job here, Elmer!  Your details descriptions show a deep level of understanding of all the concepts.  Excellent work!